smack-experimental/src/main/java/org/jivesoftware/smackx/mam/MamManager.java

```diff
@@ -250,6 +254,47 @@ public Jid getArchiveAddress() {
         return archiveAddress;
     }
 
+    /**
+     * Returns the MAM namespace used by this {@link MamManager}. If the archive does not support any MAM namespace
+     * supported by Smack, null is returned.
+     *
+     * @return the MAM namespace used by this manager, null if MAM is not supported
+     * @throws NoResponseException if there was no response from the remote entity.
+     * @throws XMPPErrorException if there was an XMPP error returned.
+     * @throws NotConnectedException if the XMPP connection is not connected.
+     * @throws InterruptedException if the calling thread was interrupted.
+     */
+    public String getMamNamespace() throws XMPPErrorException, NotConnectedException, NoResponseException, InterruptedException {
+        MamVersion mamVersion = getSupportedMamVersionOrNull();
+        return mamVersion == null ? null : mamVersion.getNamespace();
+    }
+
+    private MamVersion getSupportedMamVersionOrNull() throws XMPPErrorException, NotConnectedException, NoResponseException, InterruptedException {
+        if (mamVersion == null) {
```

Comment:
This should use the early return pattern to keep the level of indentation low.

Reply:
Good point, changed.